src/dialects/abstract/data-types.ts

```diff
@@ -0,0 +1,1826 @@
+ */
+export type FLOAT = _FLOAT;
+export const FLOAT = classToInvokable(_FLOAT);
+
+class _REAL extends FLOAT {
+  public static readonly key = 'REAL';
+}
+
+/**
+ * Floating point number (4-byte precision).
+ */
+export type REAL = _REAL;
+export const REAL = classToInvokable(_REAL);
+
+class _DOUBLE extends FLOAT {
+  public static readonly key = 'DOUBLE';
+}
+
+/**
+ * Floating point number (8-byte precision).
+ */
+export type DOUBLE = _DOUBLE;
+export const DOUBLE = classToInvokable(_DOUBLE);
+
+interface DecimalOptions extends NumberOptions {
+  scale?: number;
+  precision?: number;
+}
+
+type DecimalConstructorParams =
+  | []
+  | [DecimalOptions]
+  | [number]
+  | [number, number | undefined];
+
+class _DECIMAL extends NUMBER {
+  public static readonly key = 'DECIMAL';
+  protected _scale?: number;
+  protected _precision?: number;
+  protected options: DecimalOptions;
+
+  /**
+   * @param {string|number} [precision] defines precision
+   * @param {string|number} [scale] defines scale
+   */
+  constructor(options?: DecimalOptions);
+  constructor(precision: number, scale?: number);
+  constructor(...args: DecimalConstructorParams);
+  constructor(precision?: number | DecimalOptions, scale?: number) {
+    if (typeof precision === 'object') {
+      super(precision);
+      this.options ??= precision;
+    } else {
+      super();
+      // Sadly TS isn't smart enough to infer that this will be defined by the constructor of NUMBER.
+      // This should never do anything becuase this.options should already be defined.
+      this.options ??= {};
+
+      this.options.precision = precision;
+      this.options.scale = scale;
+    }
+
+    this._precision = this.options.precision;
+    this._scale = this.options.scale;
+  }
+
+  public toSql() {
+    if (this._precision || this._scale) {
+      return `DECIMAL(${[this._precision, this._scale]
+        .filter(num => num != null)
+        .join(',')})`;
+    }
+
+    return 'DECIMAL';
+  }
+
+  public validate(value: AcceptedNumber): asserts value is AcceptedNumber;
+  public validate(value: AcceptedNumber): true {
+    if (!Validator.isDecimal(String(value))) {
+      throw new ValidationError(
+        util.format('%j is not a valid decimal', value),
+        [],
+      );
+    }
+
+    return true;
+  }
+}
+
+/**
+ * Decimal type, variable precision, take length as specified by user
+ */
+export type DECIMAL = _DECIMAL;
+export const DECIMAL = classToInvokable(_DECIMAL);
+
+/**
+ * A boolean / tinyint column, depending on dialect
+ */
+class _BOOLEAN extends ABSTRACT<
+  boolean | Falsy,
+  boolean,
+  string | number | Buffer | boolean
+> {
+  public static readonly key = 'BOOLEAN';
+
+  public toSql() {
+    // Note: This may vary depending on the dialect.
+    return 'TINYINT(1)';
+  }
+
+  public validate(value: boolean | Falsy): asserts value is boolean;
+  public validate(value: boolean | Falsy): true {
+    if (!Validator.isBoolean(String(value))) {
```

Comment:
value says it only accepts booleans (and falsy) but this tries to also accept things like `'true` and `'false'` and whatever object stringifies to these two strings.

I think we should restrict this one to only `true` and `false` and check using `typeof value === 'boolean'`